src/config.c

```diff
@@ -801,68 +840,76 @@ void configSetCommand(client *c) {
         }
 
         /* If this config appears twice then fail */
-        for (j = 0; j < i; j++) {
-            if (set_configs[j] == config) {
-                /* Note: we don't abort the loop since we still want to handle redacting sensitive configs (above) */
-                errstr = "duplicate parameter";
-                err_arg_name = c->argv[2+i*2]->ptr;
-                invalid_args = 1;
-                break;
-            }
+        sds config_name = sdsnew(config->name);
+        if (dictFind(set_configs, config_name)) {
```

Comment:
Why do we need the `config_name` var? Why not change `optionSetDictType` to be based on a `char*` as key instead of sds? Or at least make it's hash/compare functions null terminated `char*` based?
```suggestion
        if (dictFind(set_configs, config->name)) {
```